IDE/Renesas/e2studio/RX72N/EnvisionKit_Toppers/WolfSSLDemo/src/wolfDemo/wolf_client.c

```diff
@@ -0,0 +1,177 @@
+         return (a1 << 24) | (a2 << 16) | (a3 << 8) | a4;
+    else return 0;
+}
+
+
+WOLFSSL_CTX *wolfSSL_TLS_client_init()
+{
+
+    WOLFSSL_CTX* ctx;
+#ifndef NO_FILESYSTEM
+        #ifdef USE_ECC_CERT
+        char *cert       = "./certs/ca-ecc-cert.pem";
+        #else
+        char *cert       = "./certs/ca-cert.pem";
+        #endif
+#else
+        #ifdef USE_ECC_CERT
+        const unsigned char *cert       = ca_ecc_der_256;
+        #define  SIZEOF_CERT sizeof_ca_ecc_der_256
+        #else
+        const unsigned char *cert       = ca_cert_der_2048;
+        #define  SIZEOF_CERT sizeof_ca_cert_der_2048
+        #endif
+#endif
+
+#ifdef DEBUG_WOLFSSL
+        wolfSSL_Debugging_ON();
+#endif
+
+    /* Create and initialize WOLFSSL_CTX */
+#ifdef WOLFSSL_TLS13
+    if ((ctx = wolfSSL_CTX_new(wolfTLSv1_3_client_method_ex((void *)NULL))) == NULL) 
+#else
+    if ((ctx = wolfSSL_CTX_new(wolfTLSv1_2_client_method_ex((void *)NULL))) == NULL) 
+#endif
+    {
+        printf("ERROR: failed to create WOLFSSL_CTX\n");
+        return NULL;
+    }
+
+#if !defined(NO_FILESYSTEM)
+    if (wolfSSL_CTX_load_verify_locations(ctx, cert, 0) != WOLFSSL_SUCCESS) {
+        printf("ERROR: can't load \"%s\"\n", cert);
+        return NULL;
+    }
+#else
+    if (wolfSSL_CTX_load_verify_buffer(ctx, cert, SIZEOF_CERT, WOLFSSL_FILETYPE_ASN1) != WOLFSSL_SUCCESS){
+           printf("ERROR: can't load certificate data\n");
+       return NULL;
+    }
+#endif
+
+
+    /* Register callbacks */
+    wolfSSL_SetIORecv(ctx, my_IORecv);
+    wolfSSL_SetIOSend(ctx, my_IOSend);
+    return (void *)ctx;
+
+}
+
+void wolfSSL_TLS_client(void *v_ctx, func_args *args)
+{
+    ID cepid = 3;
+    ER ercd;
+    int ret;
+    WOLFSSL_CTX *ctx = (WOLFSSL_CTX *)v_ctx;
+    WOLFSSL *ssl;
+    #define BUFF_SIZE 256
+    static const char sendBuff[]= "Hello Server\n" ;
+    char    rcvBuff[BUFF_SIZE] = {0};
+    static T_IPV4EP my_addr = { 0, 0 };
+    T_IPV4EP dst_addr;
+
+    dst_addr.ipaddr = getIPaddr(SERVER_IP);
+    dst_addr.portno = SERVER_PortNo;
+    if((ercd = tcp_con_cep(cepid, &my_addr, &dst_addr, TMO_FEVR)) != E_OK) {
+        printf("ERROR TCP Connect: %d\n", ercd);
+        return;
+    }
+
+    if((ssl = wolfSSL_new(ctx)) == NULL) {
+        printf("ERROR wolfSSL_new: %d\n", wolfSSL_get_error(ssl, 0));
+        return;
+    }
+
+    /* set callback context */
+    wolfSSL_SetIOReadCtx(ssl, (void *)&cepid);
+    wolfSSL_SetIOWriteCtx(ssl, (void *)&cepid);
+
+    if(wolfSSL_connect(ssl) != WOLFSSL_SUCCESS) {
+        printf("ERROR SSL connect: %d\n",  wolfSSL_get_error(ssl, 0));
+        return;
+    }
+
+    if (wolfSSL_write(ssl, sendBuff, strlen(sendBuff)) != strlen(sendBuff)) {
+        printf("ERROR SSL write: %d\n", wolfSSL_get_error(ssl, 0));
+        return;
+    }
+
+    if ((ret=wolfSSL_read(ssl, rcvBuff, BUFF_SIZE)) < 0) {
+        printf("ERROR SSL read: %d\n", wolfSSL_get_error(ssl, 0));
+        return;
+    }
+
+    rcvBuff[ret] = '\0' ;
```

Comment:
`ret` could be = BUFF_SIZE and would overrun `recvBuf`. Change read to `BUFF_SIZE - 1`

Reply:
Reviewed and pointed out corrections
Overflow has been addressed.